base/applications/mspaint/toolsmodel.h

```diff
@@ -68,6 +70,50 @@ struct ToolBase
     static ToolBase* createToolObject(TOOLTYPE type);
 };
 
+template <typename T>
+struct AutoPtr
```

Comment:
Do you need these auto-pointers if you use the ATL array already? (or the SimpleArray; it seems lighter)

Reply:
Polymorphism needs pointers. I cannot find auto pointers in ATL library.

Reply:
ah yes, the lists won't free the elements themselves (and you cannot use something like `CSimpleArray<ToolBase>` of course...)